### PR TITLE
Pere - Avoid boilerplate via pytorch lightning

### LINKED ISSUE #7: Logging
- Check how logging works in pytorch lightning
- Probably end up using their [WandB integration](https://docs.wandb.ai/guides/integrations/lightning) anyway (pretty graphs and dashboards ftw)

### LINKED ISSUE #8: Boilerplate
Implement lighting.ai boilerplate

### LINKED ISSUE #18: Dataset reduction
Figure out how to make a reduced version of our dataset, such that:
- Splits are deterministic. Via fixed seeds probably
- Loading is acceptably fast
- My terminal/python/VSC doesn't crash when I try to load it.

### PR DESCRIPTION
**Pull request by PereGinebra**
Implement a pytorch lightning model to make training easier. Including a simple training script for it. The model can be adapted to take more hyperparameters.

Also implemented a simple processing of the dataset. The train split is quite heavy and zipped in a single file, so unzipping it in python takes quite a while (in make_dataset.py). ~We can either all do it once and forget about it or unzip it before the script. Either way, the train set in a tensor file (.pt) takes up 27gb...~

Edit: Modified the dataset to only use the original validation set ('camelyonpatch_level_2_split_valid_x/y.h5') with a custom split for all the sets.

### CODE REVIEW

**Comment by PereGinebra** [comment]
Added WandB logging. After joining the WandB team, you will be able to see logging for all runs [here](https://wandb.ai/dtu-mlops2024-g28/Patch_Camelyon_MLOps_WandB)

**Review by MarioRojoVicente — APPROVED** [review]

**Review by kernibeni15 — APPROVED** [review]
